src/librustc_mir/build/scope.rs

```diff
@@ -221,12 +222,13 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                       extent: CodeExtent,
                       block: BasicBlock,
                       target: BasicBlock) {
-        let popped_scopes =
-            match self.scopes.iter().rev().position(|scope| scope.extent == extent) {
-                Some(p) => p + 1,
-                None => self.hir.span_bug(span, &format!("extent {:?} does not enclose",
-                                                              extent)),
-            };
+        let popped_scopes = match self.scopes
```

Comment:
another case where I think we should revert until we have some kind of heuristic around the `=`